ios/ObjectPermissions/ToDo/WelcomeViewController.swift

```diff
@@ -0,0 +1,146 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2018 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import UIKit
+import RealmSwift
+
+// Configure the permissions on the Realm and each model class.
+// This will only succeed the first time that this code is executed. Subsequent attempts
+// will silently fail due to `canSetPermissions` having already been removed.
+func initializeRealmPermissions(_ realm: Realm) {
+    // Ensure that the schema and Realm-level permissions cannot be modified by anyone but admin users.
+    let realmPermissions = realm.objects(RealmPermission.self).first!
+    let everyonePermission = realmPermissions.permissions.findOrCreate(forRoleNamed: "everyone")
+    everyonePermission.canModifySchema = false
+    everyonePermission.canSetPermissions = false
```

Comment:
Just a thing to note: Permissions are checked per-operation, so while locking down the schema care must be taken that things are done in the right order such that you still have permissions to perform the rest of the lockdown. :)

Reply:
Good point. I ran into that myself, so I guess it's worth calling out in a comment.